shell/src/main/java/org/apache/accumulo/shell/commands/FateCommand.java

```diff
@@ -130,107 +132,131 @@ public int execute(final String fullCommand, final CommandLine cl, final Shell s
-      Formatter fmt = new Formatter(buf);
-      admin.print(zs, zk, context.getZooKeeperRoot() + Constants.ZTABLE_LOCKS, fmt, filterTxid,
-          filterStatus);
-      shellState.printLines(Collections.singletonList(buf.toString()).iterator(),
-          !cl.hasOption(disablePaginationOpt.getOpt()));
-    } else if ("dump".equals(cmd)) {
-      List<Long> txids;
+    StringBuilder buf = new StringBuilder(8096);
+    Formatter fmt = new Formatter(buf);
+    admin.print(zs, zk, tableLocksPath, fmt, filterTxid, filterStatus);
+    shellState.printLines(Collections.singletonList(buf.toString()).iterator(),
+        !cl.hasOption(disablePaginationOpt.getOpt()));
+  }
 
-      if (args.length == 1) {
-        txids = zs.list();
+  private boolean deleteTx(AdminUtil<FateCommand> admin, ZooStore<FateCommand> zs,
+      ZooReaderWriter zk, ServiceLockPath zLockManagerPath, String[] args)
+      throws InterruptedException, KeeperException {
+    boolean success = true;
+    for (int i = 1; i < args.length; i++) {
+      if (admin.prepDelete(zs, zk, zLockManagerPath, args[i])) {
+        admin.deleteLocks(zk, zLockManagerPath, args[i]);
       } else {
-        txids = new ArrayList<>();
-        for (int i = 1; i < args.length; i++) {
-          txids.add(parseTxid(args[i]));
-        }
+        System.out.printf("Could not delete transaction: %s%n", args[i]);
+        return !success;
       }
+    }
+    return success;
+  }
 
-      Gson gson =
-          new GsonBuilder().registerTypeAdapter(ReadOnlyRepo.class, new InterfaceSerializer<>())
-              .registerTypeAdapter(Repo.class, new InterfaceSerializer<>())
-              .registerTypeAdapter(byte[].class, new ByteArraySerializer()).setPrettyPrinting()
-              .create();
-
-      List<FateStack> txStacks = new ArrayList<>();
+  private void validateArgs(String[] args) throws ParseException {
+    if (args.length <= 1) {
+      throw new ParseException("Must provide transaction ID");
+    }
+  }
 
-      for (Long txid : txids) {
-        List<ReadOnlyRepo<FateCommand>> repoStack = zs.getStack(txid);
-        txStacks.add(new FateStack(txid, repoStack));
+  public boolean failTx(AdminUtil<FateCommand> admin, ZooStore<FateCommand> zs, ZooReaderWriter zk,
+      ServiceLockPath managerLockPath, String[] args) {
+    boolean success = true;
+    for (int i = 1; i < args.length; i++) {
+      if (!admin.prepFail(zs, zk, managerLockPath, args[i])) {
+        System.out.printf("Could not fail transaction: %s%n", args[i]);
+        return !success;
       }
-
-      System.out.println(gson.toJson(txStacks));
-    } else {
-      throw new ParseException("Invalid command option");
     }
-
-    return failedCommand ? 1 : 0;
+    return success;
   }
 
-  protected synchronized ZooReaderWriter getZooReaderWriter(ClientContext context,
+  synchronized ZooReaderWriter getZooReaderWriter(ClientContext context,
       SiteConfiguration siteConfig, String secret) {
 
     if (secret == null) {
```

Comment:
I'm not sure why this should be synchronized. There's no state being preserved or accessed outside this method from what I can tell.